packages/block-editor/src/components/block-draggable/index.js

```diff
@@ -20,31 +22,39 @@ const BlockDraggable = ( {
 	onDragStart,
 	onDragEnd,
 } ) => {
-	const { srcRootClientId, isDraggable, icon } = useSelect(
-		( select ) => {
-			const {
-				canMoveBlocks,
-				getBlockRootClientId,
-				getBlockName,
-				getBlockAttributes,
-			} = select( blockEditorStore );
-			const { getBlockType, getActiveBlockVariation } =
-				select( blocksStore );
-			const rootClientId = getBlockRootClientId( clientIds[ 0 ] );
-			const blockName = getBlockName( clientIds[ 0 ] );
-			const variation = getActiveBlockVariation(
-				blockName,
-				getBlockAttributes( clientIds[ 0 ] )
-			);
-
-			return {
-				srcRootClientId: rootClientId,
-				isDraggable: canMoveBlocks( clientIds, rootClientId ),
-				icon: variation?.icon || getBlockType( blockName )?.icon,
-			};
-		},
-		[ clientIds ]
-	);
+	const { srcRootClientId, isDraggable, icon, getBlockListSettings } =
+		useSelect(
+			( select ) => {
+				const {
+					canMoveBlocks,
+					getBlockRootClientId,
+					getBlockName,
+					getBlockAttributes,
+					getBlockListSettings: _getBlockListSettings,
+				} = select( blockEditorStore );
+				const { getBlockType, getActiveBlockVariation } =
+					select( blocksStore );
+				const rootClientId = getBlockRootClientId( clientIds[ 0 ] );
+				const blockName = getBlockName( clientIds[ 0 ] );
+				const variation = getActiveBlockVariation(
+					blockName,
+					getBlockAttributes( clientIds[ 0 ] )
+				);
+
+				return {
+					srcRootClientId: rootClientId,
+					isDraggable: canMoveBlocks( clientIds, rootClientId ),
+					icon: variation?.icon || getBlockType( blockName )?.icon,
+					getBlockListSettings: _getBlockListSettings,
+				};
+			},
+			[ clientIds ]
+		);
+
+	const [ targetClientId, setTargetClientId ] = useState( null );
+
+	const isDropTargetValid = useIsDropTargetValid( clientIds, targetClientId );
```

Comment:
Does the `BlockDraggable` component need to know whether the drop target is valid? Could we use the `getBlockInsertionPoint` state to figure out if an insertion point is currently set instead?

Reply:
I don't think so;  given that [`getBlockInsertionPoint`](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/store/selectors.js#L1404) is always meant to return an index, it would be hard to work out which targets are valid.

Reply:
Ah, gotcha! Looks like there's another selector [`isBlockInsertionPointVisible`](https://github.com/WordPress/gutenberg/blob/e07802b3d2395c4563c6e8a8013bb2224e0927d7/packages/block-editor/src/store/selectors.js#L1442) which handles the `null` case. Could that work if we'd called `hideInsertionPoint` in the main `useCallback`?

Reply:
Oh nice find! Lemme try that out

Reply:
Ok checking if the insertion point is visible does fix the discrepancy between transparent drag chip and inserter, but I found it necessary to keep the check for valid insertion point in place, because otherwise the chip keeps flickering every time the insertion point switches places. Still it's working much better now so thanks for the tip!